tests/test_timestamp.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at:
+#
+#    http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the
+# License.
+
+# Python 2/3 compatibility
+from decimal import Decimal
+
+import pytest
+
+from amazon.ion.core import Timestamp, TimestampPrecision, record
+from tests import parametrize
+
+
+class _P(record('timestamps', 'expected_microseconds')):
+    def __str__(self):
+        return self.desc
+
+
+MISSING_MICROSECOND = _P(
+    timestamps=[Timestamp(
+            2011, 1, 1,
+            0, 0, 0, None,
+            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=0
+            ),
+            Timestamp(
+            2011, 1, 1,
+            0, 0, 0, None,
+            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=Decimal('0.123456')
+            ),
+            Timestamp(
+            2011, 1, 1,
+            0, 0, 0, None,
+            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=Decimal('0.123456789')
```

Comment:
What is the implication of `fractional_seconds` having more precision that `fractional_precision`?  This distinction needs to be documented as this surprised me...

Reply:
Currently, it's not checked whether or not the `fractional_precision` matches up to `fractional_seconds`. Should this kind of thing be done?

Reply:
It depends on why you need both--if you do, then there better be some invariant around both of these fields that is preserved (and tested).

Reply:
We don't need both anymore, as per the contract: https://github.com/amzn/ion-python/pull/100#issuecomment-537212997